### DISCUSSION TITLE
Can't Get Wireguard with WG-Dashboard working

### DISCUSSION THREAD

**Discussion by digital-knave** [community-scripts/ProxmoxVE#287] [Q&A]
I've previously installed Wireguard using the script here which works like a charm:

https://github.com/Nyr/wireguard-install

Although I do have to make the /dev/net/tun config changes as outlined in this [proxmox wiki post here](https://pve.proxmox.com/wiki/OpenVPN_in_LXC).

Of course this doesn't have the nice WG-Dashboard front end. However despite the Wireguard Helper Script from here installing just fine and WG-Dashboard working too, I just cannot get it working when I try to connect to it via my phone.

I'm guessing it must be a config issue somewhere but I've even checked the .conf files from the working version against the WG-Dashboard version and things seem set up the same so I'm struggling to see what the issue is. 

Has anyone else struggled with this or managed to get it working?

Config created from WG-Dashboard looks like this:

```
[Interface]
Address = 10.7.0.1/24
PrivateKey = [redacted]
ListenPort = 51820

[Peer]
PublicKey = [redacted]
AllowedIPs - 10.7.0.2/32
```

Peer config (which I can only really find if I select option in the peer sharing methods) looks like this:

```
[Interface]
Address = 10.7.0.2/24
DNS = 8.8.8.8
PrivateKey = [redacted]

[Peer]
PublicKey = [redacted]
AllowedIPs = 0.0.0.0/0
Endpoint = my.duckdns.domain:51820
PersistentKeepalive = 25
```
Anyone have any thoughts? For now I've just removed the LXC and gone back to the non-WGDashboard version which is working.

**Comment by MickLesk** [comment]
When you installed the script? We have merged some things 1h ago

**Reply by digital-knave** [reply]
Was earlier today. Last attempt was probably a little over an hour though

**Reply by MickLesk** [reply]
Deploy new LXC and Test again. Otherwise ask a question @ WGDashboard Project

**Reply by digital-knave** [reply]
Already asked the question over there too. I'll try the new LXC later and see if whatever you merged fixed it.

**Reply by digital-knave** [reply]
Update for you, tried again this afternoon and it still didn't work. Couple of things....

Firstly I noticed Netgear Armor on my router raising an alert when the LXC runs about a suspicious remote connection attempting to connect. Only happens with that LXC and the IP it was connecting to is 185.234.216.88 which seems to be [something weird in Russia!](https://ipinfo.io/185.234.216.88)

That aside, when I try to connect, the logs in my phone app are showing its failing at the handshake. 

Interestingly, I installed WG-Dashboard manually on the LXC that is working (as mentioned in my original post). That seems to work fine witht he existing client and peer setup. But if I create a new peer, scan the QR code with my phone and try to connect with that, it again fails at the handshake. Looking at the difference between the working version and the non-working version it seems the one that works, uses a PSK, rather than the Public/Private key combo.

Anyhow - I have WG-Dashboard working for now (with the PSK option). No idea why the pub/priv key doesn't work though.

**Reply by MickLesk** [reply]
@donaldzou / @DaanSelen

Do you or your team have any idea about this?

**Reply by DaanSelen** [reply]
Hello @digital-knave, how did you take care of the routing on the interface? normally you need to map an outgoing port to the wireguard interface. Like wg0 goes out through eth0. How did you do that?

**Reply by digital-knave** [reply]
For the one that's set up and working (which didn't use the helper-scripts and was more of a manual setup) includes a couple of lines added to the .conf file for the LXC to expose /dev/net/tun as outlined in the link in my original post. The helper-scripts docs don't say anything about this being required so I presumed it was mapping via an alternative method.

And again, for the one that's working, it seems to work just fine for a Peer with a PSK, but if I try to create a new Peer with Public/Private keys and no PSK then it won't connect.

If you want to take this over to the WG-Dashboard forum to discuss further [my Q over there is linked here](https://github.com/donaldzou/WGDashboard/discussions/503)

**Reply by rice1204** [reply]
Just wanted to reply that I'm experiencing the same exact issues. Did a fresh LXC install today but handshake fails. The only way to get it working seems to be to require PSK.

**Reply by devrimerduman** [reply]
I already installed that wireguard with wg-dashboard. I used chatgpt to have that installation scripts. if you need I can share it here.
